packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -76,7 +76,9 @@ defineProperty(Node.prototype, ShadowRootResolverKey, {
     set(this: Node, fn: ShadowRootResolver) {
         (this as any)[ShadowResolverPrivateKey] = fn;
         // TODO [#1164]: temporary propagation of the key
-        setNodeOwnerKey(this, (fn as any).nodeKey);
+        if (!isUndefined(fn)) {
```

Comment:
The typing for `fn` suggests that it can never be `undefined`, do we need to update that or is this check unnecessary?